include/oneapi/dpl/pstl/iterator_impl.h

```diff
@@ -417,6 +417,12 @@ class transform_iterator
     typedef typename ::std::iterator_traits<_Iter>::pointer pointer;
     typedef typename ::std::random_access_iterator_tag iterator_category;
 
+    transform_iterator() : __my_it_(_Iter()), __my_unary_func_(_UnaryFunc())
```

Comment:
First of all I would like to understand what use-cases we want to support. With this patch we support the default-constructor and the constructor from two arguments.

Do we think that `transform_iterator(v.begin())` use-case is worth to consider?

Reply:
> Overall. This is not enough. Previously we agreed on changing the traits check for both transform and zip_iterator to move those to common part where all the iterator traits are checked.

In this patch I am not going to unify testing of iterators traits. Not all of iterators have methods base() and functor(), so I believe this methods should be covered with separated tests.

Reply:
I think, it is, I will implement it in the next commit.